src/components/link.js

```diff
@@ -21,15 +21,15 @@ const stripPathPrefix = (path, pathPrefix) => {
 };
 
 const stripMarkdownExtension = (path) => {
-  return path.replace(/\.mdx?$/, "");
+  return path.replace(/\.mdx?(?=$|\?|#)/, "");
```

Comment:
So we're trying to catch cases of `xyz.md#` and `xyz.md?`?

Reply:
Right. Intent is that links that include the source filename continue to work after Gatsby build (e.g., possible to follow some relative links on both GitHub and in docs build). Some tests:

```
> "/xyz.mdx".replace(/\.mdx?(?=$|\?|#)/, '') // expected: /xyz
"/xyz"
> "/xyz.mdx?blah=true".replace(/\.mdx?(?=$|\?|#)/, '') // expected: /xyz?blah=true
"/xyz?blah=true"
> "/xyz.mdx#blah".replace(/\.mdx?(?=$|\?|#)/, '') // expected: /xyz#blah
"/xyz#blah"
> "/xyz.mdxtra".replace(/\.mdx?(?=$|\?|#)/, '') // expected: /xyz.mdxtra
"/xyz.mdxtra"
> "/xyz.md".replace(/\.mdx?(?=$|\?|#)/, '') // expected: /xyz
"/xyz"
> "/xyz.md?blah=true".replace(/\.mdx?(?=$|\?|#)/, '')  // expected: /xyz?blah=true
"/xyz?blah=true"
> "/xyz.md#blah".replace(/\.mdx?(?=$|\?|#)/, '') // expected: /xyz#blah
"/xyz#blah"
> "/xyz.md/blah.yaml".replace(/\.mdx?(?=$|\?|#)/, '') // expected: /xyz.md/blah.yaml
"/xyz.md/blah.yaml"
```